src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/FormatFilterExtension.java

```diff
@@ -176,47 +187,39 @@ private String toString(RuntimeObjectModel runtimeObjectModel, Object[] params,
         return "";
     }
 
-    private int getPredefinedFormattingStyleFromValue(String value) {
+    private FormatStyle getPredefinedFormattingStyleFromValue(String value) {
         switch (value.toLowerCase(Locale.ROOT)) {
             case "default":
-                return DateFormat.DEFAULT;
-            case "short":
-                return DateFormat.SHORT;
             case "medium":
-                return DateFormat.MEDIUM;
+                return FormatStyle.MEDIUM;
+            case "short":
+                return FormatStyle.SHORT;
             case "long":
-                return DateFormat.LONG;
+                return FormatStyle.LONG;
             case "full":
-                return DateFormat.FULL;
+                return FormatStyle.FULL;
             default:
-                return -1;
+                return null;
         }
     }
 
-    private String formatDate(String format, Date date, Locale locale, TimeZone timezone) {
+    private String formatDate(String format, Date date, Locale locale, TimeZone timezone, DateTimeFormatter formatter) {
         if (date == null) {
             return null;
         }
         try {
-            final DateFormat formatter;
-            int formattingStyle = getPredefinedFormattingStyleFromValue(format);
-            if (formattingStyle != -1) {
-                if (locale != null) {
-                    formatter = DateFormat.getDateInstance(formattingStyle, locale);
+            if (formatter == null) {
```

Comment:
where is this private method called with formatter != null? Why do we need that parameter?

Reply:
This was a leftover of a wip commit.